api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,82 @@ message JwtHeader {
   string value_prefix = 2;
 }
 
+// This message specifies a Jwt requirement.
+//
+// .. code-block:: yaml
+//
+//    required_issuers: [jwt_issuer1, jwt_issuer2]
+//
+// PerRouteConfig for jwt_authn filter uses this message.
+message JwtRequirement {
+  // Specify list of required Jwt issuers. If list is empty,  Jwt is NOT required.
+  // If any of required issuers is missing, or its verification fails, the request will
+  // be rejected.
+  // If a Jwt issuer is not required and is in the config, it will be verified.
+  // But it will not fail the request if its verification fails.
+  repeated string required_issuers = 1;
+}
+
+// This message specifies a Jwt requirement for a specific Route condition.
+// For example:
+//
+// .. code-block:: yaml
+//
+//    - match:
+//         prefix: "/healthz"
+//      requirement:
+//         required_issuers: []
+//    - match:
+//         prefix: "/"
+//      requirement:
+//         required_issuers: [jwt_issuer1]
+//         allow_missing_or_failed: true
+//
+message RouteRequirement {
+  // The route matching parameter.  Only match is satisfied, the other fields will apply.
+  //
+  // For example: following match will match all requests.
+  //
+  // .. code-block:: yaml
+  //
+  //    match:
+  //       prefix: "/"
+  //
+  envoy.api.v2.route.RouteMatch match = 1 [(validate.rules).message.required = true];
+
+  // Specifies the Jwt requiement for matched routes.
+  JwtRequirement requirement = 2;
+}
+
 // This is the Envoy HTTP filter config for JWT authentication.
 // [#not-implemented-hide:]
 message JwtAuthentication {
   // List of JWT rules to valide.
   repeated JwtRule rules = 1;
 
+  // DEPRECATED: Please use "routes" field. if "routes" field is not empty,
+  // this field will not be used.
   // If true, the request is allowed if JWT is missing or JWT verification fails.
   // Default is false, a request without JWT or failed JWT verification is not allowed.
   bool allow_missing_or_failed = 2;
 
-  // This field lists the patterns allowed to bypass JWT verification. This only applies when
-  // `allow_missing_or_failed_jwt` is false. Under this config, if a request doesn't have JWT, it
-  // will be rejected. But some requests still needed to be forwarded without JWT, such as OPTIONS
-  // for CORS and some health checking paths.
+  // Specifies requirements based on the route matches. The first matched requirement will be
+  // applied. If there are overlapped match conditions, please put the most specific match first.
+  // If PerRouteConfig is specified for the route, this field will not be uesd.
   //
-  // Examples: bypass all CORS options requests
```

Comment:
What's the plan with other kinds of request matching (e.g. header matching for CORS options requests)?

Reply:
envoy.api.v2.route.RouteMatch supports header matching in additional to path matching.